x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/util/DateUtils.java

```diff
@@ -105,7 +111,11 @@ public static ZonedDateTime asDateTime(String dateFormat) {
     }
 
     public static ZonedDateTime ofEscapedLiteral(String dateFormat) {
-        return ZonedDateTime.parse(dateFormat, DATE_TIME_ESCAPED_LITERAL_FORMATTER.withZone(UTC));
+        try {
```

Comment:
This could be improved by checking whether there's a `T` separator at its dedicated position in order to select the right parser. This should avoid the fallback in case of an exception.

Reply:
The position is not dedicated though as the year can be more than 4 digits.
But I can use `lastIndexOf('-')`